README.md

```diff
@@ -1,35 +1,251 @@
+   * It must have a binary entry point without any parameters.
+
+2. Write a patterns.json with the configuration of your tool.
+    * This file must be located on /docs/patterns.json.
+      * **name:** Unique identifier of the tool (lower-case letters without spaces)
+      * **patterns:** The patterns that the tool provides
+          * **patternId:** Unique identifier of the pattern (lower-case letters without spaces)
+          * **level:** Severity level of the issue 
+          * **category:** Category of the issue 
+          * **parameters:** Parameters received by the pattern
+            * **name:** Unique identifier of the parameter (lower-case letters without spaces)
+            * **default:** Default value of the parameter
+    ```json
+    {
+      "name":"jshint",
+      "patterns":[
+        {
+          "patternId":"latedef",
+          "level": "Error",
+          "category": "UnusedCode",
+          "parameters":[
+            {
+              "name":"latedef",
+              "default":"vars"
+            }
+          ]
+        }
+      ]
+    }
+    ```
+    #### Levels and Categories
+    For level types we have:
+    * Error, Warning, Info
+    
+    For category types we have:
+    * ErrorProne, CodeStyle, UnusedCode, Security, Compatibility, Performance, Documentation
+
+3. Write the code to run the tool
+You don't have to use this seed and you can write the code in any language you want but, you have to invoke the tool according to the configuration.
+After you have your results from the tool, you should print them to the standard output in our Result format, one result per line.
+    * The filename should **not** include the prefix "/src/". Example: 
+        * absolute path: /src/folder/file.js
+        * filename path: folder/file.js
+    ```json
+    {
+      "filename":"codacy/core/test.js",
+      "message":"found this in your code",
+      "patternId":"latedef",
+      "line":2
+    }
+    ```
+    * If you are not able to run the analysis for any of the files requested you should return an error for each one of them to the standard output in our Error format.
+    ```json
+    {
+      "filename":"codacy/core/test.js",
+      "message":"could not parse the file"
+    }
+    ```
+
+## Tool Documentation
+At Codacy we strive to provide the best value to our users and, to accomplish that, we document our patterns so that the user can better understand the problem and fix it.
+
+At this point, your tool has everything it needs to run, but there is one other really important thing that you should do before submitting your docker: the documentation for your tool.
 
-[Scala Docker Template Docs](http://docs.codacy.com/v1.5/docs/tool-developer-guide-using-scala)
+Your files for this section should be placed in /docs/description/.
+
+In order to provide more details you can create:
+* A single /docs/description/description.json
+* **(Optional)** A /docs/description/<PATTERN-ID>.md for each pattern. 
```

Comment:
Question, do we still want this to be optional?

Reply:
I theory it can be, but we want to have it as  much as possible, so lets omit the optional.

Reply:
ok, will remove it from here.